### PR TITLE
Adding new process rawTDStoSignal

### PR DESCRIPTION
**Pull request by juanangp**
juanangp Large: 277 [](https://gitlab.cern.ch/rest-for-physics/rawlib/-/commits/rawTDStoSignal) [](https://gitlab.cern.ch/rest-for-physics/framework/-/commits/rawTDStoSignal)

New feature, new process added to transform TDS (Tektronix oscilloscope DAQ) data format (https://lfna.unizar.es/gifna/tds) into rawSignalEvent. The different oscilloscope channels are stored in consecutive channel ids from 0 to 4 if present.

### CODE REVIEW

File: src/TRestRawTDSToSignalProcess.cxx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+/// Inherits from TRestRawToSignalProcess: 
+/// * **pedScript**: Not used
+/// * **runScript**: Not used
+/// * **electronics**: You need to write here TDS
+/// * **fileFormat**: Not used
+///
+/// ### Examples
+/// Similar to TRestRawToSignalProcess, jus use TDS as electronics parameter
+/// \code
+///   <addProcess type="TRestRawTDSToSignalProcess" name="virtualDAQ" value="ON">
```
Comment (jgalan) [inline src/TRestRawTDSToSignalProcess.cxx:39]: If other parameters are not required we could just skip them and write the process definition in a simpler form.

```
<addProcess type="TRestRawTDSToSignalProcess" name="tdsAQS" electronics="TDS"/>
```

Comment (juanangp) [reply src/TRestRawTDSToSignalProcess.cxx]: Indeed, you are right, only `electronics` is a required parameter.
Just updated the description.

File: src/TRestRawTDSToSignalProcess.cxx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+/// produced by the read binary files produced with TDS (Tektronix oscilloscope) DAQ,
+/// The results are stores following TRestRawSignalEvent format ,and then it 
+/// processing can continue inside with the REST framework libraries.
+///
+/// ### Parameters
+/// Inherits from TRestRawToSignalProcess, only electronics type is required: 
+/// * **electronics**: You need to write here TDS
+///
+/// ### Examples
+/// Similar to TRestRawToSignalProcess, jus use TDS as electronics parameter
```
Comment (jgalan) [inline src/TRestRawTDSToSignalProcess.cxx:34]:
```suggestion
/// Similar to TRestRawToSignalProcess, we just define the `electronics` parameter as `TDS`
```

**Comment by jgalan** [comment]
I just added an example to attempt to read a TDS file. I have created the examples directory that it will be installed under `REST_PATH/examples/raw`. In fact, I recently added the `examples` path to the default search paths, then we can just run them from anywhere. See commit rest-for-physics/framework@652fd26 at the draft PR rest-for-physics/framework#198.

So, if I compile using the branch `jgalan_masks` from framework, and this branch at rawlib, and I try to test as follows, I get into troubles:

```
jgalan@sultan:~$ restManager --c tdsAQS.rml --f /storage/iaxo/RyD/gem-mm/GEMMM0034.raw.01 
Warning in <UnknownClass::SetDisplay>: DISPLAY not set, setting it to 79.116.70.207:0.0
Starting pid: 39763
                            
              [==                                  Launching TRestManager...                                   ==]              
                            
              [==                                    Creating TRestManager                                     ==]              
              [==                                            path:.                                            ==]              
terminate called after throwing an instance of 'std::runtime_error'
  what():  popen() failed!
Aborted
jgalan@sultan:~$ 

```

**Comment by juanangp** [comment]
> restManager --c tdsAQS.rml --f /storage/iaxo/RyD/gem-mm/GEMMM0034.raw.01

That's weird, I have tried locally and it runs properly:
image

**Comment by jgalan** [comment]
Ok, it also does for me now 8/

**Review by jgalan — APPROVED** [review]

**Pull request merged**